src/Response/SapiEmitterTrait.php

```diff
@@ -10,26 +10,28 @@
 namespace Zend\Diactoros\Response;
 
 use Psr\Http\Message\ResponseInterface;
+use RuntimeException;
 
 trait SapiEmitterTrait
 {
     /**
-     * Inject the Content-Length header if is not already present.
+     * Checks to see if content has previously been sent.
      *
-     * @param ResponseInterface $response
-     * @return ResponseInterface
+     * If either headers have been sent or the output buffer contains content,
+     * raises an exception.
+     *
+     * @throws RuntimeException if headers have already been sent.
+     * @throws RuntimeException if output is present in the output buffer.
      */
-    private function injectContentLength(ResponseInterface $response)
+    private function assertNoPreviousOutput()
     {
-        if (! $response->hasHeader('Content-Length')) {
-            // PSR-7 indicates int OR null for the stream size; for null values,
-            // we will not auto-inject the Content-Length.
-            if (null !== $response->getBody()->getSize()) {
-                return $response->withHeader('Content-Length', (string) $response->getBody()->getSize());
-            }
+        if (headers_sent()) {
+            throw new RuntimeException('Unable to emit response; headers already sent');
         }
 
-        return $response;
+        if (ob_get_level() > 0 && ob_get_length() > 0) {
```

Comment:
@mindplay-dk: note that I separated out this from the `headers_sent()` check; I did this to allow having discretely different exception messages.

Reply:
line 32: the PR is called 'Remove output buffer awareness', but line 32 is checking the output buffer :)

Reply:
Technically, so is calling `headers_sent()`. :)

I suppose I should retitle it "remove output buffer flushing"...

Reply:
Re-named the PR to "Removes Content-Length injection and output buffer flushing".